homeassistant/components/media_player/channels.py

```diff
@@ -0,0 +1,395 @@
+"""
+Support for interfacing with an instance of Channels.
+https://getchannels.com
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.channels/
+"""
+import logging
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    MEDIA_TYPE_CHANNEL, MEDIA_TYPE_TVSHOW, MEDIA_TYPE_EPISODE,
+    MEDIA_TYPE_VIDEO, SUPPORT_PLAY, SUPPORT_PAUSE, SUPPORT_STOP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_NEXT_TRACK, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_PLAY_MEDIA, SUPPORT_SELECT_SOURCE, DOMAIN, MediaPlayerDevice)
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT, STATE_IDLE, STATE_PAUSED, STATE_PLAYING,
+    STATE_UNKNOWN, ATTR_ENTITY_ID)
+
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+FEATURE_SUPPORT = SUPPORT_PLAY | SUPPORT_PAUSE | SUPPORT_STOP | \
+    SUPPORT_VOLUME_MUTE | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_SELECT_SOURCE
+
+DATA_CHANNELS = 'channels'
+TIMEOUT = 1
+
+SERVICE_SEEK_FORWARD = 'channels_seek_forward'
+SERVICE_SEEK_BACKWARD = 'channels_seek_backward'
+SERVICE_SEEK_BY = 'channels_seek_by'
+
+# Service call validation schemas
+ATTR_SECONDS = 'seconds'
+
+CHANNELS_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+CHANNELS_SEEK_BY_SCHEMA = CHANNELS_SCHEMA.extend({
+    vol.Required(ATTR_SECONDS): vol.Coerce(int),
+})
+
+
+class Channels(object):
```

Comment:
Please extract this into it's own PyPi package.

Reply:
✅ 

https://github.com/fancybits/pychannels